File: src/main/java/me/fzzyhmstrs/amethyst_core/interfaces/KillTracking.java
package me.fzzyhmstrs.amethyst_core.interfaces;

import net.minecraft.entity.LivingEntity;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.server.world.ServerWorld;

public interface KillTracking {

    default void onWearerKilledOther(ItemStack stack, LivingEntity wearer, LivingEntity victim, ServerWorld world){

    }

    default void incrementKillCount(ItemStack stack){
        NbtCompound nbt = stack.getOrCreateNbt();
        if (!nbt.contains("kills")){
            nbt.putInt("kills",1);
        } else {
            int kills = nbt.getInt("kills");
            nbt.putInt("kills",kills + 1);
        }
    }

}

File: src/main/kotlin/me/fzzyhmstrs/amethyst_core/item_util/interfaces/KillTracking.kt
package me.fzzyhmstrs.amethyst_core.item_util.interfaces

import net.minecraft.entity.LivingEntity
import net.minecraft.entity.damage.DamageSource
import net.minecraft.item.ItemStack
import net.minecraft.server.world.ServerWorld

interface KillTracking {
    fun onWearerKilledOther(stack: ItemStack, wearer: LivingEntity, victim: LivingEntity, world: ServerWorld)
}